core/src/main/java/com/sequenceiq/cloudbreak/job/stackpatcher/ExistingStackPatcherJob.java

```diff
@@ -65,21 +69,30 @@ protected void executeTracedJob(JobExecutionContext context) throws JobExecution
             String stackPatchTypeName = context.getJobDetail().getJobDataMap().getString(STACK_PATCH_TYPE_NAME);
             StackPatchType stackPatchType = stackPatchTypeConverter.convertToEntityAttribute(stackPatchTypeName);
             if (stackPatchType == null || StackPatchType.UNKNOWN.equals(stackPatchType)) {
-                unscheduleJob(context);
-                throw new JobExecutionException(String.format("Stack patch type %s is unknown", stackPatchTypeName));
-            }
-            Optional<ExistingStackPatchService> optionalExistingStackPatchService = getStackPatchServiceForType(stackPatchType);
-            if (optionalExistingStackPatchService.isEmpty()) {
-                unscheduleJob(context);
-                throw new JobExecutionException("No stack patcher implementation found for type " + stackPatchType);
+                String message = String.format("Stack patch type %s is unknown", stackPatchTypeName);
+                unscheduleAndFail(message, context, stack, stackPatchType);
+            } else {
```

Comment:
is there any reason why `else` statements added to these logics? as I see those are not needed

Reply:
for readability reasons, as they are not inline throws that exit the function early

Reply:
makes sense